lua/apidocs/snacks.lua

```diff
@@ -26,10 +26,18 @@ local function get_data_dirs(opts)
     return { data_dir }
   end
   local dirs = {}
+  local install_dirs = vim.fn.readdir(data_dir)
   for _, source in ipairs(opts.restrict_sources) do
-    local dir = data_dir .. source .. "/"
-    if vim.fn.isdirectory(dir) == 1 then
-      table.insert(dirs, dir)
+    for _, entry in ipairs(install_dirs) do
+      if vim.fn.isdirectory(data_dir .. entry) == 1 then
+        if entry == source then
+          table.insert(dirs, data_dir .. entry .. "/")
+        elseif entry:match("^" .. source .. "~%d+(%.%d+)?$") then
+          -- check for partial match
+          -- e.g. "python" -> "python~3.12"
+          table.insert(dirs, data_dir .. entry .. "/")
```

Comment:
similar to apidocs.lua, maybe just add an elseif for the original if, and make the loop only if the original if fails. and then in the loop we can only do the match, as we know equality won't work.